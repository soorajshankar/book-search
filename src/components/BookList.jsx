import React from 'react'
import ListItem from './ListItem'

const BookList = ({list = []}) => (
  <div className="c-list">
    {list.length > 0 ? (
      list.map((book, i) => (
        <ListItem
          key={i}
          summary={book.summary}
          title={book.title}
          author={book?.author?.author}
        />
      ))
    ) : (
      <p style={{textAlign: 'center'}}>No books Selected</p>
    )}
  </div>
)
export default BookList
