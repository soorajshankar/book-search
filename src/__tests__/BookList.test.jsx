import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import BookList from '../components/BookList'

describe('renders BookList', () => {
  it('should match Snapshot', () => {
    let list = [
      {author: {author: 'author1'}, summary: 'summary1', title: 'title'},
    ]
    const {getByText, container} = render(<BookList list={list} />)
    expect(container).toMatchSnapshot()
  })
  it('should render with values', () => {
    let list = [
      {author: {author: 'author1'}, summary: 'summary1', title: 'title'},
    ]
    const {getByText, container} = render(<BookList list={list} />)
    expect(container.querySelector('.c-card__title').textContent).toEqual(
      'title',
    )
    expect(container.querySelector('.c-card__author').textContent).toEqual(
      'author1',
    )
    expect(container.querySelector('.c-card__summary').textContent).toEqual(
      'summary1',
    )
  })
})
