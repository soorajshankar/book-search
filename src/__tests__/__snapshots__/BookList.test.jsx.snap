// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BookList should match Snapshot 1`] = `
<div>
  <div
    class="c-list"
  >
    <div
      class="c-list__item"
    >
      <h3
        class="c-card__title"
      >
        title
      </h3>
      <h4
        class="c-card__author"
      >
        author1
      </h4>
      <p
        class="c-card__summary"
      >
        summary1
      </p>
    </div>
  </div>
</div>
`;
